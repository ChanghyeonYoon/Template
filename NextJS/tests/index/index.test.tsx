// tests/index/index.test.ts

import { render, RenderResult } from "@testing-library/react";
import Home from "pages/index";

describe("Home", () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(<Home />);
  });

  it("should render the component", () => {
    expect(component).toBeDefined();
  });

  it("renders Test Home", () => {
    const test = component.getByText("Kongtare - Malma");
    expect(test).toBeInTheDocument();
  });

  afterEach(() => {
    expect(component.container).toMatchSnapshot();
    component.unmount();
  });
});
