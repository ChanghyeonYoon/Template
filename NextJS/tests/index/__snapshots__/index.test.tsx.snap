// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Test Home 1`] = `
<div>
  <main>
    <div>
      <h1>
        Kongtare - Malma
      </h1>
    </div>
  </main>
</div>
`;

exports[`Home should render the component 1`] = `
<div>
  <main>
    <div>
      <h1>
        Kongtare - Malma
      </h1>
    </div>
  </main>
</div>
`;
